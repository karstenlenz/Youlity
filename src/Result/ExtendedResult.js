import PropTypes from 'prop-types'
import React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components/macro'
import Button from '../common/Button'
import HeadlineUnderline from '../common/HeadlineUnderline'
import SectionBGWithButton from '../common/SectionBGWithButton'
import TextLink from '../common/TextLink'
import { capitalizeFirstLetter } from '../common/util'
import { personalityStyleData } from '../data/personalityStyleData'
import ResultBar from './ResultBar'
import useResults from './useResults'

ExtendedResult.propTypes = {
  questionnaireIds: PropTypes.string.isRequired,
  results: PropTypes.string.isRequired,
}

export default function ExtendedResult({ questionnaireIds, results }) {
  const resultData = useResults(questionnaireIds, results)
  const sortedResultData = resultData
    .slice()
    .sort((a, b) => b.yesCount - a.yesCount)
  return (
    <>
      <img src="/img/style_info.svg" alt="" />
      <HeadlineUnderline>
        <h1>Ergebnis</h1>
      </HeadlineUnderline>
      <IntroText>
        Sie haben bisher folgende Fragebögen ausgefüllt. Klicken Sie auf einen
        Namen, um zu den Detail-Informationen zu gelangen.
      </IntroText>
      <section>
        {sortedResultData.map((result, index) => (
          <Link to={'/style-info/' + result.id} key={result.id}>
            <ResultSection>
              <ResultHeadline>
                {capitalizeFirstLetter(
                  personalityStyleData[result.id - 1]?.adjective
                )}{' '}
                {'>'}
              </ResultHeadline>
              <ResultBar
                percentage={(result.yesCount / 8) * 100}
                index={index}
              />
            </ResultSection>
          </Link>
        ))}
      </section>

      {9 - results.length > 0 ? (
        <>
          {9 - results.length === 1 ? (
            <p>Es gibt noch 1 weiteren Fragebogen.</p>
          ) : (
            <p>Es gibt noch {9 - results.length} weitere Fragebögen.</p>
          )}
          <Link to={'/questionnaire/entry/' + questionnaireIds + '/' + results}>
            <Button>Weiteren Fragebogen starten</Button>
          </Link>
          <SectionBGWithButton>
            <h2>Tagebuch</h2>
            <p>
              Sind Sie manchmal unzufrieden mit Ihrer Persönlichkeit? Unsere
              Tagebuch-Funktion hilft Ihnen, Ihr Verhalten zu analysieren und
              dadurch erste Schritte zu einer Veränderung anzustoßen.
            </p>
            <Link to="/journal">
              <Button btnType="secondary">Zum Tagebuch</Button>
            </Link>
          </SectionBGWithButton>
        </>
      ) : (
        <>
          <p>
            Sie haben die Fragebögen zu allen 9 Persönlichkeitsstilen
            ausgefüllt. Wir hoffen, dass Sie etwas Nützliches über Ihre
            Persönlichkeit mitnehmen konnten.
            <br />
            Wenn Sie noch mehr über sich erfahren wollen, nehmen Sie am besten
            Kontakt zu einem Psychotherapeuten auf.
          </p>
          <TextLinkWithMargin href="https://www.wege-zur-psychotherapie.org/">
            Infoseite "Wege zur Psychotherapie" &gt;
          </TextLinkWithMargin>
          <SectionBGWithButton>
            <h2>Tagebuch</h2>
            <p>
              Sind Sie manchmal unzufrieden mit Ihrer Persönlichkeit? Unsere
              Tagebuch-Funktion hilft Ihnen, Ihr Verhalten zu analysieren und
              dadurch erste Schritte zu einer Veränderung anzustoßen.
            </p>
            <Link to="/journal">
              <Button btnType="secondary">Zum Tagebuch</Button>
            </Link>
          </SectionBGWithButton>
          <BottomSpacer />
        </>
      )}
    </>
  )
}

const IntroText = styled.p`
  margin-bottom: 15px;
`
const TextLinkWithMargin = styled(TextLink)`
  margin: 15px 0;
`

const BottomSpacer = styled.div`
  margin-bottom: 40px;
`

const ResultSection = styled.section`
  margin-bottom: 20px;
`

const ResultHeadline = styled.h3`
  margin: 0 0 5px 0;
  text-align: left;
`
