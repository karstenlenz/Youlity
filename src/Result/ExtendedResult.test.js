import { render } from '@testing-library/react'
import React from 'react'
import ExtendedResult from './ExtendedResult'
import { BrowserRouter as Router } from 'react-router-dom'

describe('The ExtendedResult component', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Router>
        <ExtendedResult questionnaireIds="1928373645" results="6422" />
      </Router>
    )
    expect(container).toMatchSnapshot()
  })
})
