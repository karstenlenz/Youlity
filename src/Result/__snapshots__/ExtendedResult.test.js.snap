// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ExtendedResult component renders correctly 1`] = `
<div>
  <img
    alt=""
    src="/img/style_info.svg"
  />
  <section
    class="HeadlineUnderline__HeadlineContainer-sc-1reeb6v-1 tAJUt"
  >
    <div
      class="HeadlineUnderline__ChildContainer-sc-1reeb6v-0 cbzWsz"
    >
      <h1>
        Ergebnis
      </h1>
    </div>
    <svg
      class="HeadlineUnderline__UnderlineStyled-sc-1reeb6v-2 dVDSEH"
    >
      underline.svg
    </svg>
  </section>
  <p
    class="ExtendedResult__IntroText-sc-1w1w0fq-0 fcwGt"
  >
    Sie haben bisher folgende Fragebögen ausgefüllt. Klicken Sie auf einen Namen, um zu den Detail-Informationen zu gelangen.
  </p>
  <section>
    <a
      href="/style-info/1"
    >
      <section
        class="ExtendedResult__ResultSection-sc-1w1w0fq-3 lfKtj"
      >
        <h3
          class="ExtendedResult__ResultHeadline-sc-1w1w0fq-4 jCwgvR"
        >
          Selbstbewusst
           
          &gt;
        </h3>
        <div
          class="ResultBar__Outline-sc-479x3f-1 echTYD"
        >
          <div
            class="ResultBar__ColorBar-sc-479x3f-2 MirAr"
          >
            <p
              class="ResultBar__Text-sc-479x3f-0 bCfkyT"
            >
              deutlich ausgeprägt
            </p>
          </div>
        </div>
      </section>
    </a>
    <a
      href="/style-info/9"
    >
      <section
        class="ExtendedResult__ResultSection-sc-1w1w0fq-3 lfKtj"
      >
        <h3
          class="ExtendedResult__ResultHeadline-sc-1w1w0fq-4 jCwgvR"
        >
          Sprunghaft
           
          &gt;
        </h3>
        <div
          class="ResultBar__Outline-sc-479x3f-1 echTYD"
        >
          <div
            class="ResultBar__ColorBar-sc-479x3f-2 kuxa-dL"
          >
            <p
              class="ResultBar__Text-sc-479x3f-0 bCfkyT"
            >
              wenig ausgeprägt
            </p>
          </div>
        </div>
      </section>
    </a>
    <a
      href="/style-info/2"
    >
      <section
        class="ExtendedResult__ResultSection-sc-1w1w0fq-3 lfKtj"
      >
        <h3
          class="ExtendedResult__ResultHeadline-sc-1w1w0fq-4 jCwgvR"
        >
          Gesellig
           
          &gt;
        </h3>
        <div
          class="ResultBar__Outline-sc-479x3f-1 echTYD"
        >
          <div
            class="ResultBar__ColorBar-sc-479x3f-2 LsDfm"
          >
            <p
              class="ResultBar__Text-sc-479x3f-0 bCfkyT"
            >
              Nicht ausgeprägt
            </p>
          </div>
        </div>
      </section>
    </a>
    <a
      href="/style-info/8"
    >
      <section
        class="ExtendedResult__ResultSection-sc-1w1w0fq-3 lfKtj"
      >
        <h3
          class="ExtendedResult__ResultHeadline-sc-1w1w0fq-4 jCwgvR"
        >
          Wachsam
           
          &gt;
        </h3>
        <div
          class="ResultBar__Outline-sc-479x3f-1 echTYD"
        >
          <div
            class="ResultBar__ColorBar-sc-479x3f-2 bekMn"
          >
            <p
              class="ResultBar__Text-sc-479x3f-0 bCfkyT"
            >
              Nicht ausgeprägt
            </p>
          </div>
        </div>
      </section>
    </a>
  </section>
  <p>
    Es gibt noch 
    <strong>
      5
       weitere
    </strong>
     
    Fragebögen.
  </p>
  <a
    href="/questionnaire/entry/1928373645/6422"
  >
    <button
      class="Button__BtnStyled-sc-129yfz5-0 iWqmsN"
      width="100"
    >
      Weiteren Fragebogen starten
    </button>
  </a>
  <section
    class="SectionBGWithButton__Background-sc-11viaht-0 llKPXe"
  >
    <h2>
      Tagebuch
    </h2>
    <p>
      Sind Sie manchmal unzufrieden mit Ihrer Persönlichkeit? Unsere Tagebuch-Funktion hilft Ihnen, Ihr Verhalten zu analysieren und dadurch erste Schritte zu einer Veränderung anzustoßen.
    </p>
    <a
      href="/journal"
    >
      <button
        class="Button__BtnStyled-sc-129yfz5-0 cxViIx"
        width="100"
      >
        Zum Tagebuch
      </button>
    </a>
  </section>
</div>
`;
